seamare/src/error.rs

```diff
@@ -0,0 +1,14 @@
+use std::path::PathBuf;
+use thiserror::Error;
+
+pub type Result<T, E = SeamareError> = std::result::Result<T, E>;
+
+#[derive(Error, Debug)]
+pub enum SeamareError {
+    #[error("Loading path failed")]
+    IO(#[from] std::io::Error),
```

Comment:
I'm a little worried about these errors, can we remove the `From` impl from the first one and add a `.map_err(SeamareError::IO)` wherever necessary? ?

```suggestion
    IO(#[source] std::io::Error),
```